lib/biran/configurinator.rb

```diff
@@ -30,12 +30,17 @@ def files_to_generate
         .tap { |files_list| files_list.each(&sanitize_config_files(files_list)) }
     end
 
-    def create(name:, extension:, output_dir: nil, output_name: nil)
+    def create(name:, extension:, output_dir: nil, output_name: nil, config_index_list: [])
       output_dir ||= config_dir
       output_name ||= name
       generated_file = ERBConfig.new(filtered_config, name, extension, config_dir, output_dir, output_name)
       generated_file.bindings = bindings
-      generated_file.save!
+      return generated_file.save! unless config_index_list.any?
```

Comment:
This keeps current behavior.

Reply:
Is `output_name` used if `config_index_list` is present?

Reply:
Yes - we set output_name above and use `name` to find the template based on that name.
If you look below this line, output name is modified to add the config index list number to the end of the name.
```
generated_file.output_name = "#{output_name}-#{config_index}"
```
That will set the output name to include the index. If there is no index list, we still use output name. If you are using this config index feature, you are asking to generate a set of files from one template with names following this convention. If you want to use a specific name, then you would need to use a specific block like before, and setup a separate template. 
Think of this like a factory, where we are creating multiple files from the same source, with slightly different internals, they can't all have the same name or they would over write each other on generation.

Reply:
Right, yeah I see it now. Thanks